public/app/plugins/panel/table/module.tsx

```diff
@@ -75,10 +81,76 @@ export const plugin = new PanelPlugin<PanelOptions, PanelFieldConfig>(TablePanel
     },
   })
   .setPanelOptions((builder) => {
-    builder.addBooleanSwitch({
-      path: 'showHeader',
-      name: 'Show header',
-      description: "To display table's header or not to display",
-      defaultValue: defaultPanelOptions.showHeader,
-    });
+    builder
+      .addBooleanSwitch({
+        path: 'showHeader',
+        name: 'Show header',
+        description: "To display table's header or not to display",
+        defaultValue: defaultPanelOptions.showHeader,
+      })
+      .addRadio({
+        category: ['Table footer'],
+        path: 'footerMode',
+        name: 'Mode',
+        settings: {
+          options: [
+            { label: 'None', value: 'none' },
+            // { label: 'auto', value: 'auto' },  // TODO: when frame meta supports this
+            { label: 'Summary', value: 'summary' },
+            { label: 'Provided', value: 'frame' },
+          ],
+        },
+        defaultValue: 'none',
+      })
+      .addCustomEditor({
+        category: ['Table footer'],
+        id: 'reduceOptions.calcs',
```

Comment:
should this be on: footerSummary.reducer?

Reply:
i wasn't sure what id did.  it sets a reference to the model?

Reply:
updated